File: library/src/main/java/com/afollestad/iconrequest/SendResult.java
package com.afollestad.iconrequest;

import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class SendResult {

  static SendResult create(int sentCount, boolean usedPolarRm) {
    return new AutoValue_SendResult(sentCount, usedPolarRm, null);
  }

  static SendResult create(Exception e) {
    return new AutoValue_SendResult(0, false, e);
  }

  public abstract int sentCount();

  public abstract boolean usedPolarRm();

  public boolean success() {
    return error() != null;
  }

  @Nullable
  public abstract Exception error();
}

File: library/src/main/java/com/afollestad/iconrequest/SendResult.kt
package com.afollestad.iconrequest

/** @author Aidan Follestad (afollestad) */
data class SendResult(
  val sentCount: Int = 0,
  val usedArcticRm: Boolean = false,
  val error: Throwable? = null
) {
  val success: Boolean
    get() = error == null
}
